homeassistant/components/solarlog/const.py

```diff
@@ -17,83 +30,202 @@
-        "consumptionMONTH",
-        "consumption month",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:power-plug",
-    ],
-    "consumption_year": [
-        "consumptionYEAR",
-        "consumption year",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:power-plug",
-    ],
-    "consumption_total": [
-        "consumptionTOTAL",
-        "consumption total",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:power-plug",
-    ],
-    "total_power": ["totalPOWER", "total power", "Wp", "mdi:solar-power"],
-    "alternator_loss": [
-        "alternatorLOSS",
-        "alternator loss",
-        POWER_WATT,
-        "mdi:solar-power",
-    ],
-    "capacity": ["CAPACITY", "capacity", PERCENTAGE, "mdi:solar-power"],
-    "efficiency": [
-        "EFFICIENCY",
-        "efficiency",
-        f"% {POWER_WATT}/{POWER_WATT}p",
-        "mdi:solar-power",
-    ],
-    "power_available": [
-        "powerAVAILABLE",
-        "power available",
-        POWER_WATT,
-        "mdi:solar-power",
-    ],
-    "usage": ["USAGE", "usage", None, "mdi:solar-power"],
-}
+
+@dataclass
+class SolarLogSensorEntityDescription(SensorEntityDescription):
+    """Sensor entity description for Solarlog."""
+
+    json_key: str | None = None
+
+
+SENSOR_TYPES = [
+    SolarLogSensorEntityDescription(
+        key="time",
+        json_key="TIME",
+        name="last update",
+        icon="mdi:calendar-clock",
+        device_class=DEVICE_CLASS_TIMESTAMP,
+    ),
+    SolarLogSensorEntityDescription(
+        key="power_ac",
+        json_key="powerAC",
+        name="power AC",
+        icon="mdi:solar-power",
+        unit_of_measurement=POWER_WATT,
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SolarLogSensorEntityDescription(
+        key="power_dc",
+        json_key="powerDC",
+        name="power DC",
+        icon="mdi:solar-power",
+        unit_of_measurement=POWER_WATT,
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SolarLogSensorEntityDescription(
+        key="voltage_ac",
+        json_key="voltageAC",
+        name="voltage AC",
+        icon="mdi:flash",
+        unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+        device_class=DEVICE_CLASS_VOLTAGE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SolarLogSensorEntityDescription(
+        key="voltage_dc",
+        json_key="voltageDC",
+        name="voltage DC",
+        icon="mdi:flash",
+        unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+        device_class=DEVICE_CLASS_VOLTAGE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SolarLogSensorEntityDescription(
+        key="yield_day",
+        json_key="yieldDAY",
+        name="yield day",
+        icon="mdi:solar-power",
+        unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_MEASUREMENT,
+        last_reset=dt.start_of_local_day(),
```

Comment:
This is already added to `yield_total`, any specific reason for adding it to this one as well?

Reply:
No, no specific reason, I was under the impression that all sensor that start measure in the past and update now needed this `last_reset`. But for the energy component, I assume only one is fine (one for consumption, one for solar production). And than the overall production would make the most sense, I agree. Removed this one.